src/Features/CSharp/Portable/ConvertAnonymousTypeToTuple/CSharpConvertAnonymousTypeToTupleCodeFixProvider.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToTuple
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertAnonymousTypeToTupleCodeFixProvider)), Shared]
+    internal class CSharpConvertAnonymousTypeToTupleCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.ConvertAnonymousTypeToTupleDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAllWithEditorAsync(context.Document,
+                    e => FixInCurrentMember(context.Document, e, context.Diagnostics[0], c), c)),
+                context.Diagnostics);
+
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private async Task FixInCurrentMember(
+            Document document, SyntaxEditor editor,
+            Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            // For the standard invocation of the code-fix, we want to fixup all creations of the
```

Comment:
I don't know what happens with anonymous types in lambda or local functions. When are they considered "same" types (ie. to be replaced) or different?

Reply:
I didn't hear back from you on this question

---
In reply to: [190789240](https://github.com/dotnet/roslyn/pull/27099#discussion_r190789240) [](ancestors = 190789240)

Reply:
I went with: 

"they should be viewed as the same type, and we'd want to change them all from the highest method's span inwards."

However, i didn't do tests.  I will add them now :)

Reply:
Done.